electron/main.ts

```diff
@@ -552,48 +553,73 @@ interface Tools {
   wine: string
 }
 
+const runWineTricks = async (
+  prefix: string,
+  wine: string,
+  isProton: boolean
+) => {
+  const winetricks = `${heroicToolsPath}/winetricks`
+  const newProtonWinePath = wine.replace(
+    new RegExp('proton' + '$'),
+    'files/bin/wine64'
+  )
+  const oldProtonWinePath = wine.replace(
+    new RegExp('proton' + '$'),
+    'dist/bin/wine64'
+  )
+
+  const protonWinePath = existsSync(newProtonWinePath.replaceAll("'", ''))
+    ? newProtonWinePath
+    : oldProtonWinePath
+
+  const wineBin = isProton ? protonWinePath : wine
+
+  let winePrefix = prefix.replace('~', userHome)
+  if (isProton) {
+    const protonPrefix = winePrefix.replaceAll("'", '')
+    winePrefix = `${protonPrefix}/pfx`
+  }
+
+  const command = `WINEPREFIX='${winePrefix}' WINE='${wineBin}' ${winetricks} -q`
+
+  logInfo(['trying to run', command], LogPrefix.Backend)
+  try {
+    const { stderr, stdout } = await execAsync(command, execOptions)
+    logInfo(`Output: ${stderr} \n ${stdout}`)
+  } catch (error) {
+    logError(`${error}`)
+    logError(
+      `Something went wrong! Check if WineTricks is available and ${wineBin} exists`,
+      LogPrefix.Backend
+    )
+  }
+}
+
 ipcMain.handle(
   'callTool',
   async (event, { tool, wine, prefix, exe }: Tools) => {
-    const newProtonWinePath = wine.replace('proton', 'files/bin/wine64')
-    const oldProtonWinePath = wine.replace('proton', 'dist/bin/wine64')
-    const isProton = wine.includes('proton')
-    const winetricks = `${heroicToolsPath}/winetricks`
+    const isProton = wine.includes('/proton')
+
+    if (tool === 'winetricks') {
+      return runWineTricks(prefix, wine, isProton)
+    }
+
     const options: ExecOptions = {
       ...execOptions
     }
 
-    // existsSync is weird because it returns false always if the path has single-quotes in it
-    const protonWinePath = existsSync(newProtonWinePath.replaceAll("'", ''))
-      ? newProtonWinePath
-      : oldProtonWinePath
-    let wineBin = isProton ? `'${protonWinePath}'` : wine
-    let winePrefix = prefix.replace('~', userHome)
+    const wineBin = isProton ? `'${wine}' run` : wine
```

Comment:
Ideally all of this should use `runWineCommand`, to avoid having this logic twice.

Problem with that is that we don't know the AppName of the game we're running this on. If we did, we could make this a lot simpler. Not sure how easy that is to implement in the frontend

Reply:
shoulds be one more argument only, I will test a few things